ball_detect/main.c

```diff
@@ -107,7 +108,8 @@ ISR(PCINT0_vect) {
 				starts_of_pulses[ct] = TCNT1; 
 			} else {
 				ends_of_pulses[ct] = TCNT1;
-				lenghts_of_pulses[ct] = abs(ends_of_pulses[ct] - starts_of_pulses[ct]);
+				lenghts_of_pulses[ct] = abs(ends_of_pulses[ct] -
+											starts_of_pulses[ct]);
```

Comment:
This line looks terrible with hard tabs. Please set ts=8 noet and align it that way using spaces as well.